src/Workspaces/Core/Portable/Workspace/Solution/Solution.CompilationTracker.cs

```diff
@@ -274,7 +274,7 @@ public CompilationTracker FreezePartialStateWithTree(Solution solution, Document
                 }
 
                 inProgressProject = inProgressProject.AddProjectReferences(newProjectReferences);
-                if (!Enumerable.SequenceEqual(inProgressCompilation.References, metadataReferences))
+                if (!Enumerable.SequenceEqual(inProgressCompilation.References, metadataReferences, MetadataReferenceComparer.Instance))
```

Comment:
Rather than patching up the issue here, can we just not do the work above that caused it in the first place?